### DISCUSSION TITLE
Sometimes doesn´t power on and sometimes makes a red light before booting up completely

### DISCUSSION THREAD

**Discussion by alerivo** [iRobotEducation/create3_docs#393] [Questions]
### How are you connecting to your Create 3?

Wi-Fi and USB Ethernet (ROS 2)

### Computer(s) Model(s) and Operating System(s)

Intel NUC with Ubuntu Server 22.04 is connected to the robot

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

H.1.2

### Which RMW is your robot running?

FastRTPS

### Is the robot connected to a network? If so, what is the network type?

Robot is connected via WiFi to home netwok and via Ethernet (using a USB-C hub) to the Intel NUC

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

None

### What is the Adapter Board's USB/BLE Toggle currently switched to?

USB

### Describe your question.

As the title introduces, we are having, sometimes, problems turning on the robot.

The robot started to being used with H.1.0, maybe with that version or H.1.1, after connecting for the first time the Create3 with the Intel NUC via Ethernet, the robot started to make a red light before it completely boots up. Now the robot is using H.1.2.
The steps are:
The robot is powered off.
After placing the robot on the dock the ring shows a white light making circles.
Some seconds later the ring goes completely red.
And some time later the robot connects to the WiFi, rings a chime, for some seconds the ring goes solid green/yellow and then solid white.

This problem isn't that annoying, but it takes more time that it should to boot up and it shouldn't go through the red light booting up.

Another problem that it happens from time to time and started to do after using the Ethernet for first time too (so both problems could be related) is that the robot directly doesn't power on.
The steps are:
The robot is powered off.
The robot is placed on the dock.
And it doesn't turns on! While some green and orange leds blinks in the middle of the robot the ring doesn´t turns on.

At some point the robot just turns on.
Different things were tried when this happens like disconnect Ethernet and cable power of the Intel NUC, move the robot so it makes better contact with the Dock, unplug and plug de Dock, etc.
Nothing really helped.
Waiting till the next day to try again or letting the robot on the Dock till powers on have been the way to "solve" this problem.

Similar discussions have been searched, #183 have been found. The problem discussed there is only the first one, related to the red light, although it seems that robot never connected to WiFi while this one it connects to WiFi at some point.

The red light problem happens almost always while the not powering on problem doesn't happen that often.

The following system info and logs are after booting up making the red light. It didn't the second error this time.

---
 system uptime
 19:32:54 up 23 min,  load average: 1.07, 1.23, 1.02

 system memory
              total        used        free      shared  buff/cache   available
Mem:          59080       35812        7508         204       15760       21372
Swap:             0           0           0

 system ifconfig
eth0      Link encap:Ethernet  HWaddr C8:4D:44:20:B1:71  
          inet addr:192.168.186.2  Bcast:192.168.186.255  Mask:255.255.255.0
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:1471 errors:0 dropped:1 overruns:0 frame:0
          TX packets:1997 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:129600 (126.5 KiB)  TX bytes:330928 (323.1 KiB)

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:705 errors:0 dropped:0 overruns:0 frame:0
          TX packets:705 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:170400 (166.4 KiB)  TX bytes:170400 (166.4 KiB)

wlan0     Link encap:Ethernet  HWaddr 50:14:79:44:BB:0D  
          inet addr:192.168.0.4  Bcast:192.168.0.255  Mask:255.255.255.0
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:38116 errors:0 dropped:110 overruns:844 frame:844
          TX packets:769 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:11370220 (10.8 MiB)  TX bytes:472272 (461.2 KiB)
          Interrupt:6 

---

[Logs](https://github.com/iRobotEducation/create3_docs/files/11579149/messages.txt)

**Comment by alerivo** [comment]
It just happened that didn't power on being on the dock.

[This](https://github.com/iRobotEducation/create3_docs/assets/1330344/53072524-f5dc-4b0c-81d0-836904820ea8) video shows how it behaves in this situations.

After some minutes powered on, going trough the red light.

[This](https://github.com/iRobotEducation/create3_docs/files/11580150/messages.txt) are the logs.

Thanks for any help/advice you could give!

**Reply by justinIRBT** [reply]
The log you attached shows a standard bootup sequence for the create.  The application restarts when the robot gets an IP address, as the ROS2 DDS layer doesn't dynamically reconfigure when new interfaces come online.  The log shows the application startup, OS gets a wlan IP address, restarts the application, then it shows "ready to go" at the end.

Regarding your power issues, I haven't used the NUC with the robot, but I wonder if it is at the edge of the robot's power budget.  If the NUC is drawing more power that the charger is providing, the battery can drain until it hits its safety protection.  I wonder if you need to charge your robot with the NUC turned off to get the battery fully charged and ideally, shutdown the NUC when the robot returns to charge.  If you try this workflow do you still run into power issues?

If you have a scenario where the red light comes on after the robot has been running for many minutes (after the 2 application starts bootup described above), can you attach a log of that scenario?

**Reply by alerivo** [reply]
Hello justin thank you for your answer.

So the red light while it restarts to boot for second time due to the new IP address is expected? It should do the red light while rebooting?

I can't remember a situation that the Create3 did the red light after it has completely booted up and been running for some minutes. Anyway, in case it happens, I will attach such logs.

About the power issues if I am not wrong there has been times where the robot didn't power on being fully charged and with the NUC completely unplugged from it. I will try to follow your workflow. Anyway sometimes is needed to keep using the robot and NUC so it gets charged with everything running. I can assure the charging level increases in this cases.

Thank you again!

**Comment by alerivo** [comment]
Hi! The other day it happened that it wasn't powering on while being on the dock.
[These](https://github.com/iRobotEducation/create3_docs/files/11929166/messages.txt) are the logs after it boot.
